### PR TITLE
Cors route plugin

### PR DESCRIPTION
**Pull request by mitchdraft**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1020

### CODE REVIEW

File: projects/gloo/pkg/plugins/consul/subsets.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
 )
 
-func (p *plugin) ProcessRouteAction(params plugins.RouteParams, inAction *v1.RouteAction, inPlugins map[string]*plugins.RoutePlugin, out *envoyroute.RouteAction) error {
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/consul/subsets.go]: what is this change?

File: projects/gloo/pkg/plugins/cors/plugin.go [hunk 2]
```diff
@@ -17,6 +25,16 @@ type plugin struct {
 
 var _ plugins.Plugin = new(plugin)
 var _ plugins.HttpFilterPlugin = new(plugin)
+var _ plugins.RoutePlugin = new(plugin)
+
+var (
+	InvalidDualSpecError    = errors.New("invalid cors spec: must specify one of VirtualHostPlugins.Cors or CorsPolicy (deprecated) - both were provided")
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/cors/plugin.go]: if both are provided, maybe use the new one instead of erroring?

File: projects/gloo/pkg/plugins/plugin_interface.go [hunk 3]
```diff
@@ -54,9 +54,11 @@ type RoutePlugin interface {
 	ProcessRoute(params RouteParams, in *v1.Route, out *envoyroute.Route) error
 }
 
+// note: any route action plugin can be implemented as a route plugin
+// suggestion: if your plugin requires configuration from a RoutePlugin field, implement the RoutePlugin interface
 type RouteActionPlugin interface {
 	Plugin
-	ProcessRouteAction(params RouteParams, inAction *v1.RouteAction, inPlugins map[string]*RoutePlugin, out *envoyroute.RouteAction) error
+	ProcessRouteAction(params RouteParams, inAction *v1.RouteAction, out *envoyroute.RouteAction) error
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/plugin_interface.go]: why this change?

File: projects/gloo/pkg/plugins/cors/plugin.go (hunk 2 above)
Comment (mitchdraft) [reply projects/gloo/pkg/plugins/cors/plugin.go]: ok, how about logging a warning

File: projects/gloo/pkg/plugins/plugin_interface.go (hunk 3 above)
Comment (mitchdraft) [reply projects/gloo/pkg/plugins/plugin_interface.go]: that arg is never used
I don't think we want to support it either - requires making a map of all the plugins. I think it's better to implement anything that would use that construct as a route plugin

File: projects/gloo/pkg/plugins/cors/plugin.go (hunk 2 above)
Comment (yuval-k) [reply projects/gloo/pkg/plugins/cors/plugin.go]: sounds good

File: projects/gloo/pkg/plugins/plugin_interface.go (hunk 3 above)
Comment (yuval-k) [reply projects/gloo/pkg/plugins/plugin_interface.go]: ok

File: projects/gloo/pkg/plugins/consul/subsets.go (hunk 1 above)
Comment (mitchdraft) [reply projects/gloo/pkg/plugins/consul/subsets.go]: same as below
